src/main/java/executor/command/CommandConvert.java

```diff
@@ -0,0 +1,194 @@
+     */
+    @Override
+    public void execute(Wallet wallet) {
+        Double convertedAmount = this.convertCurrency(this.getFrom(), this.getTo(), this.getAmount());
+        Ui.dukeSays(this.result(convertedAmount) + rateUsed());
+        Ui.printSeparator();
+    }
+
+
+    @Override
+    public void execute(TaskList taskList) {
+
+    }
+
+
+    /**
+     * extractAmount parses the user input from CLI to get the amount which the user wishes to convert.
+     * @param commandType this is the type of command
+     * @param userInput takes the user entered input from CLI
+     * @return the amount which user wishes to convert is returned
+     */
+    private Double extractAmount(CommandType commandType, String userInput) {
+        String amountStr = Parser.parseForPrimaryInput(commandType, userInput);
+        return Double.parseDouble(amountStr);
+    }
+
+    /**
+     * getFromUserInput parses user input for the flag "from".
+     * @param userInput this is the user entered input from CLI
+     * @return this function returns the 3 character unique string
+     */
+    private String getFromUserInput(String userInput) {
+        String fromStr = Parser.parseForFlag("from", userInput);
+        return fromStr;
+    }
+
+    /**
+     * getToUserInput parses the user input for the flag "to".
+     * @param userInput this is the user entered input from CLI
+     * @return this function return the 3 character unique string
+     */
+    private String getToUserInput(String userInput) {
+        String toStr = Parser.parseForFlag("to", userInput);
+        return toStr;
+    }
+
+    /**
+     * getTheJson is a function which gets the JSON object from the api.
+     * @param from this is the country from which we are converting currency
+     * @param to this is the country to which we are converting the currency
+     * @return this function returns a string version of the json object or else it will return null
+     */
+    private String getTheJson(String from, String to) {
+        try {
+            URL url = new URL("https://api.ratesapi.io/api/latest?base=" + from + "&symbols=" + to);
+            BufferedReader br = new BufferedReader(new InputStreamReader(url.openStream()));
+            String str = "";
+            String completeJson = "";
+            while (null != (str = br.readLine())) {
+                completeJson += str;
+            }
+            return completeJson;
+        } catch (Exception ex) {
+            ex.printStackTrace();
+            return null;
+        }
+    }
+
+    /**
+     * convertCurrency is a function used to loop through the string version of the json.
+     * @param from this is the country from which we are converting currency
+     * @param to this is the country to which we are converting the currency
+     * @param amount this is amount of currency which the user wants to convert
+     * @return this function returns the converted amount of the currency or else returns null
```

Comment:
Mention the @param type i.e.

@param from String representing the country code to convert the currency from.

Reply:
Refactored as requested